src/book/02-system/01-nubots/02-build-system.mdx

```diff
@@ -2,6 +2,472 @@
+|   -u   | The user to install to on the target. Defaults to the user in the Docker image (this default user should almost always be correct for our robots) |
+|   -t   | Install toolchain to the target                                                                                                                   |
+|  -cn   | Only install new config files. This is the default                                                                                                |
+|  -cu   | Update config files on the target that are older than the local files                                                                             |
+|  -co   | Overwrite all config files on the target                                                                                                          |
+|  -ci   | Ignore all changes to config files (installs no config files)                                                                                     |
+
+The toolchain is a collection of 3rd party libraries that have been built into the Docker images. On a new robot, all roles will fail to run unless the toolchain is installed and, more often than not, if you are receiving errors on the robot saying something to the effect of "cannot open shared object file: No such file or directory" this will usually be solved by installing the toolchain.
+
+<Alert type='info'>
+
+If you have installed the toolchain and you are still getting errors like the
+one mentioned above, try running `sudo ldconfig` on the robot.
+
+</Alert>
+
+##### Module
+
+The `module` command allows you to generate a new module. To generate a new line detection module one would run
+
+```bash
+./b module generate vision/LineDetector
+```
+
+This will generate all necessary module files in the `module/vision/LineDetector` folder. This includes an empty configuration file, a basic C++ source and header file implementing the `LineDetector` reactor, and an empty unit test file.
+
+##### NBS
+
+NBS (NUClear Binary Stream) files are, effectively, concatenations of protobuf messages with a small header packet per message. They provide a simple and effective means for recording real data from the robots in real time.
+
+The `nbs` command provides a couple of subcommands to help you work with NBS data.
+
+| Subcommand     | Description                                                                              |
+| -------------- | ---------------------------------------------------------------------------------------- |
+| extract_images | Extracts images from the NBS file and save them to disk                                  |
+| json           | Extracts selected messages from the NBS file and dumps them to `stdout` in a json format |
+| stats          | Calculates message statistics on the provided NBS file                                   |
+| video          | Extract images from the NBS file and save them in a video file using FFmpeg              |
+
```

Comment:
calibrate_cameras will also live in here, so when https://github.com/NUbots/NUbots/pull/279 gets merged in this needs to be updated

Reply:
yep, that can be added in a later PR